machine-learning/app/main.py

```diff
@@ -61,6 +63,12 @@ async def lifespan(_: FastAPI) -> AsyncGenerator[None, None]:
         gc.collect()
 
 
+async def preload_models(cache: ModelCache, preload_models: list[tuple[ModelType, str]]):
```

Comment:
The cache is a global variable so no need to make it an arg here

Reply:
I thought that it would be easier to test but yeah you're right that it's not necessary, I now used `monkeypatch` in tests to set the `model_cache` in `app.main`